narciuch/assignment4/src/IslandFinder.java

```diff
@@ -0,0 +1,132 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class IslandFinder {
+
+  /**
+   * Returns the number of islands in the lands matrix.
+   *
+   * <p>Two tiles in the matrix are part of the same island if
+   * they are both land and are adjacent horizontally or vertically.
+   *
+   * @param nrRows    number of rows in the lands matrix
+   * @param nrColumns number of columns in the lands matrix
+   * @param lands     nested boolean array, with true representing land,
+   *                  false representing water
+   * @return the number of islands in the lands matrix
+   */
+  public int getIslands(int nrRows, int nrColumns, boolean[][] lands) {
```

Comment:
Yes, I completely agree. I did this because in the specification it was written that "The input to your function is the number of rows, number of columns, and a 2-dimensional array of booleans, where false means water and true means land." I will change the code t just extract row and column numbers from array itself as this definitely makes more sense.